mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs

```diff
@@ -235,12 +235,11 @@ where
     let addr_format = builder.address_format()?;
     let my_address = AddressBuilder::new(
         addr_format,
-        AddressHashEnum::AddressHash(key_pair.public().address_hash()),
         conf.checksum_type,
         conf.address_prefixes.clone(),
         conf.bech32_hrp.clone(),
     )
-    .as_pkh()
+    .as_pkh_from_pk(*key_pair.public())
```

Comment:
Here we are adding support for p2pk for DerivationMethod::SingleAddress.
I guess we need a TODO somewhere to add the same for HDWallet when it is merged

Reply:
However, in utxo_withdraw.rs we create an Address with pubkey inside, for withdraw from a HD account. That is, assuming for HD accounts we should provide P2PK support

Reply:
> However, in utxo_withdraw.rs we create an Address with pubkey inside, for withdraw from a HD account. That is, assuming for HD accounts we should provide P2PK support

Yeah I should probably revert that to avoid confusion. p2pk should never be used in an HD account.

Reply:
> Is this done?

The issue was in `utxo_withdraw.rs`, we were creating an address that has the plain pubkey (thus can query p2pk address) even though we don't support that.
Looking at this snippet now, this seems to be the same issue as well, `priv_key_policy` could be an `HDWallet` so we shouldn't use `as_pkh_from_pk` and use `as_pkh` instead when it's the case. I am confused though why we use `DerivationMethod::SingleAddress` with an `HDWallet`?

> should HD PRs be merged first?

Yeah go ahead and do that.